docs/04-10-architecture.md

```diff
@@ -3,37 +3,39 @@
+Runtime Watcher is configured and deployed in a Kyma cluster in the Kyma reconciliation loop.
 
-## Workflow
+### Listener package
 
-The diagram presents Runtime Watcher's workflow.
+The Listener package is designed to streamline the process of establishing an endpoint for an operator located in KCP. This operator intends to receive Watcher Events that are transmitted from Runtime Watcher to KCP. When calling the `RegisterListenerComponent` function, it returns you a runnable listener, which should be added to your reconile-manager, and a channel. See this [example of how the listener package is used in Lifecycle Manager](https://github.com/kyma-project/lifecycle-manager/blob/24d21bb642ceaf9dadffe7732bf7c3f70c085ffb/controllers/manifest_controller.go#L43-L50). The channel becomes the source for the operator. For example, the operator can fetch the incoming WatchEvents from the channel and requeue the corresponding resource in the reconcile loop. Furthermore, it is possible to provide a validation function to the `RegisterListenerComponent` which can be used to filter out not needed requests before processing them further. An example of the validation function is SAN pinning.
 
-![Runtime Watcher architecture](./assets/runtime_watcher_architecture_simplified.svg)
+To learn how to set up a listener, read the [Listener document](./Listener.md).
 
-KCP environment includes multiple Watcher CRs that are reconciled by Lifecycle Manager. It means that Lifecycle Manager configures VirtualServices on KCP. <!--TBD: update the diagram to multiply the VirtualService-->
+#### Subject Alternative Name (SAN) pinning
 
-Istio Gateway is static and the Lifecycle Manager service is attached to the Lifecycle Manager deployment. VirtualServices are configured dynamically by Watcher CRs.
+[SAN pinning](https://github.com/kyma-project/lifecycle-manager/blob/c1e06b7b973aca17cc715b6a4660b76f4e7b9e29/pkg/security/san_pinning.go#L55) is an example implementation of a validation function given to the listener package. SAN pinning is used in Lifecycle Manager. The validation function checks if the certificate of the incoming WatchEvent request has at least one matching SAN with the domain of the corresponding Kyma CR. The domain of a Kyma CR is saved in an annotation called `skr-domain`.
 
-Runtime Watcher and certificates <!--TBD: or certificate Secret--> are installed in every Kyma reconciliation in a Kyma runtime.
+### Certificates
+Since it is required to have an mTLS connection from the Kyma cluster to the KCP Gateway, signed TLS certificates are needed for both the KCP Gateway and each Kyma cluster. In order not to rely on third parties in the infrastructure, we build the `Public Key Infrastructure`(PKI) by [bootstraping a self-signed CA issuer](https://cert-manager.io/docs/configuration/selfsigned/#bootstrapping-ca-issuers). The PKI uses the function of the [Cert-Manager](https://cert-manager.io/). 
```

Comment:
- Can we switch to more active voice?
- No "we" according to our [style guide](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/02-style-and-terminology.md#pronouns).

```suggestion
Because you must have an mTLS connection from the Kyma cluster to the KCP Gateway,  you need signed TLS certificates for both the KCP Gateway and each Kyma cluster. To be independent from third parties in the infrastructure, Kyma built the `Public Key Infrastructure`(PKI) by [bootstraping a self-signed CA issuer](https://cert-manager.io/docs/configuration/selfsigned/#bootstrapping-ca-issuers). The PKI uses the function of the [Cert-Manager](https://cert-manager.io/). 
```